client_code/portable/_register.py

```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2021 anvilistas
+
+from anvil.server import SerializationError
+
+from anvil_extras.utils import import_module
+
+__version__ = "0.0.1"
+
+registered_types = {}
+registered_names = {}
+
+
+def register(cls, name=None):
+    if name is None:
+        name = cls.__module__ + "." + cls.__name__
+    registered_types[cls] = name
+    registered_names[name] = cls
+    return cls
+
+
+def get_registered_cls(tp_name):
+    try:
+        return registered_names[tp_name]
+    except KeyError:
+        pass
+
+    mod, _ = tp_name.rsplit(".", 1)
+    import_module(mod)
+    try:
+        return registered_names[tp_name]
```

Comment:
I don't get why this line would behave any differently to when it was tried on L24. I'm probably missing something here.

Reply:
If we send a serialziable type to the server it may not have been imported into server code
since it exists in client code

So we try to import the module
We assume that when the module gets imported the registration occurs inside that module
which means the second go should we should have the registered class

Reply:
> We assume that when the module gets imported the registration occurs inside that module

Ah! That's the bit I was missing!